include/engine/guidance/assemble_steps.hpp

```diff
@@ -25,10 +25,13 @@ namespace guidance
 namespace detail
 {
 StepManeuver stepManeuverFromGeometry(extractor::guidance::TurnInstruction instruction,
-                                      const WaypointType waypoint_type,
                                       const LegGeometry &leg_geometry,
-                                      const std::size_t segment_index,
-                                      const unsigned exit);
+                                      const std::size_t segment_index);
```

Comment:
From what I can see I _think_ you only need the instruction to pass it on (and you do not move it), so a const-ref (or const value since it should be cheap to copy it) should be fine, too.